### PR TITLE
Don't create duplicate issues when configuration is invalid

### PR DESCRIPTION
**Pull request by csomh**
Kudos to @ppisar for reporting this :man_singer: 

Signed-off-by: Hunor Csomortáni <csomh@redhat.com>

### CODE REVIEW

**Review by jpopelka — APPROVED** [review]
good catch!

**Comment by csomh** [comment]
Please hold the horses! I would like to add some tests.

**Review by mfocko — APPROVED** [review]
damn that suffix idea... :upside_down_face:

**Comment by csomh** [comment]
One more round of review please! Here is what changed:
- Changed the way `create_issue_if_needed` works. There is a more explicit `title` arg now, which should simplify things and the only thing that is added in the method is a `[packit] ` prefix.
- Added a few unit tests for this method, just to be sure :stuck_out_tongue_winking_eye:

File: tests/unit/test_config.py [hunk 1]
```diff
@@ -335,3 +335,49 @@ def test_get_package_config_from_repo_not_found_exception_nonexisting_issue():
...
+def test_create_issue_if_needed(issues, create_new, title, message):
+    project = flexmock()
+    check = lambda value: value is None  # noqa
+    project.should_receive("get_issue_list").and_return(issues).once()
+
+    if create_new:
+        project.should_receive("create_issue").with_args(
+            title=f"[packit] {title}", body=message
+        ).and_return(flexmock(name="new issue")).once()
+        check = lambda value: value.name == "new issue"  # noqa
```
Comment (mfocko) [inline tests/unit/test_config.py:380]:
```suggestion
        ).and_return(flexmock(title="new issue")).once()
        check = lambda value: value.title == "new issue"  # noqa
```

**Review by mfocko — APPROVED** [review]
LGTM :+1:

**Comment by lachmanfrantisek** [comment]
Thanks!

**Pull request merged**